UIP-0027.md

```diff
@@ -0,0 +1,93 @@
+# UIP-27: Transaction service fields
+
+```
+Author:   Dima Saveliev <dima@thirdhash.com>
+Status:   Draft
+Created:  2019-05-07
+```
+
+## Abstract
+
+  This document describes protocol changes in the transaction structure:
+the `nVersion` field is going to be replaced by four separate service fields:
+* type
+* version
+* two more fields are vacant
+
+
+## Motivation
+
+  [UIP-0018](https://github.com/dtr-org/uips/blob/master/UIP-0018.md) introduced **a transaction type** concept 
+and added the corresponding field as a part of already existing `nVersion` field. 
+That was achieved by overloading `nVersion` with bit arithmetic at the cost of increasing complexity of the code.
+
+  By implementing separate service fields we solve the following problems:
+  
+  1. Clean-up the logic behind the `type` and `version` fields.
+  2. Minimize the size of `type` and `version` fields.
+  3. Minimize the impact on the existing test suite.
+  
+  As an additional benefit, we will have two more fields for possible future needs.
+  
+  
+## Specification
+
+At the moment we have this `nVersion` field structure:
+ 
+_These pictures illustrate `nVersion` in big-endian order as we have it in the runtime. 
+It's worth noting, however, that the serialized bytes will have a little-endian order._
+
+```
+  0x00      0x00     0x00      0x00   
+^------^  ^------^  ^----------------^
+ unused     type         version      
+^------------------------------------^
+             nVersion
+```
+
+* `nVersion` - original `uint32_t` value
+* `type` - derived `uint8_t` attribute
+* `version` - derived `uint16_t` attribute
+
+Proposed layout for the same bytes:
+```
+  0x00      0x00      0x00      0x00     
+^------^  ^------^  ^------^  ^------^   
+reserved    type    reserved  version    
+```
+
+* `type` - independent`uint8_t` field
+* `version` - independent`uint8_t` field
+
+
+## Rationale
```

Comment:
In both `Rationale` and `Motivation` parts, I did not find a description of a real problem we're trying to solve. The essential proposed change is literally to shrink version from 16 bytes to 8 bytes that leads to having two dead bytes instead of one.

Personally, I'd rather replace nVersion (4 bytes) with two fields: version and txtype (each 1 byte). As for bitcoin tests which relies on "real" transactions, I think we can get rid of them in favor of new tests for unit-e transactions. Having this compatibility with bitcoin limits us in further transactions manipulations.

If we say that it's too hard to achieve and maintain (which is understandable), then I don't see a point in this UIP. We can pinch unused byte of nVersion anytime we need, as we already did it in UIP-18.